File: done/sentimental-readability/readability.py
def main():
    text = input("Input: ")
    L, W, S = analyse(text)
    index = round(0.0588 * (100.0 * L / W) - 0.296 * (100.0 * S / W) - 15.8)
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def analyse(str):
    L = S = 0
    W = 1
    for i in str:
        if i.isalpha():
            L += 1
        elif i == " ":
            W += 1
        elif i == "." or i == "!" or i == "?":
            S += 1
    return L, W, S


main()

File: done/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

string text;
int L = 0;
int W = 1;
int S = 0;
int analyse(string str);

int main(void)
{
    // get text
    text = get_string("Input: ");

    // analyse
    analyse(text);

    // formula
    int index = round(0.0588 * (100.0 * L / W) - 0.296 * (100.0 * S / W) - 15.8);

    // print
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int analyse(string str)
{
    int len = strlen(str);
    for (int i = 0; i < len; i++)
    {
        char word = str[i];
        if (isalpha(word))
        {
            L++;
        }
        else if (word == ' ')
        {
            W++;
        }
        else if (word == '.' || word == '!' || word == '?')
        {
            S++;
        }
    }
    return 0;
}